ldes-server-compaction/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/compaction/domain/services/CompactableRelationPredicate.java

```diff
@@ -0,0 +1,30 @@
+package be.vlaanderen.informatievlaanderen.ldes.server.compaction.domain.services;
+
+import be.vlaanderen.informatievlaanderen.ldes.server.fragmentation.entities.Fragment;
+import be.vlaanderen.informatievlaanderen.ldes.server.fragmentation.repository.FragmentRepository;
+
+import java.util.function.Predicate;
+
+public class CompactableRelationPredicate implements Predicate<Fragment> {
+
+	private final FragmentRepository fragmentRepository;
+
+	public CompactableRelationPredicate(FragmentRepository fragmentRepository) {
+		this.fragmentRepository = fragmentRepository;
+	}
+
+	@Override
+	public boolean test(Fragment fragment) {
+		return firstFragmentHasOneOutGoingRelation(fragment) &&
+				secondFragmentHasOneIncomingRelation(fragment);
+	}
+
+	private boolean secondFragmentHasOneIncomingRelation(Fragment fragment) {
+		return fragmentRepository.retrieveFragmentsByOutgoingRelation(fragment.getRelations().get(0).treeNode())
```

Comment:
Is it correct to take the first relation here? Since the second fragment is based on the max predicate and determined in "retrieveMostCompactedRelationFragment". I would expect that the second fragment determined in PaginationCompactionService is supposed to have one outgoing relation?

Why don't we pass first and second fragment as parameter for the test above?

Reply:
Although, it's correct I think the condition can be a bit stricter. Updated